core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/TransformCorrelatedSingleRowSubqueryToProject.java

```diff
@@ -70,8 +72,19 @@ public Result apply(CorrelatedJoinNode parent, Captures captures, Context contex
             }
         }
 
-        if (subquery instanceof ValuesNode values && isSingleRowValuesWithNoColumns(values)) {
-            return Result.ofPlanNode(parent.getInput());
+        if (subquery instanceof ValuesNode values) {
+            if (isSingleRowValuesWithNoColumns(values)) {
+                return Result.ofPlanNode(parent.getInput());
+            }
+            if (values.getRowCount() == 1 && values.getRows().orElseThrow().get(0) instanceof Row row) {
```

Comment:
nit: should you reflect changes in class javadoc?

Reply:
The javadoc was still valid. I added another example with subquery containing no projection.